Removing deprecated validator signature (#123)
wayfair-incubator/columbo: docs/usage-guide/validators.md

```

> [^1]:

Reviewer: There are three, total references to this footnote in this [file](https://github.com/wayfair-incubator/columbo/blob/ef72659f086229a7853d381ad64ddf841f67ff5a/docs/usage-guide/validators.md#validator-structure). I recommend removing the first two references to this footnote as they are no longer relevant.

Reviewer 2: I feel like the footnote can be removed entirely. Then rephrase the `Upgrading Validator Structure` section to mentioned that "`Optional[str]` was used <0.10.0 and was removed on 1.0.0"

Reviewer: Good point - since the docs are versioned, we can remove this footnote and section entirely.

Author: Cool updated to remove

>     Formerly, the type alias for a `Validator` was `Callable[[str, Answers], Union[ValidationResponse, Optional[str]]]` - the difference
>     from the documentation above is that a `Validator` can return either a `ValidationResponse` *or* `Optional[str]`. Returning `Optional[str]`

Reviewer: Suggested improvement:

```suggestion
    Formerly, the type alias for a `Validator` was `Callable[[str, Answers], Union[ValidationResponse, Optional[str]]]` -
    in which a `Validator` can return either a `ValidationResponse` *or* `Optional[str]`. Returning `Optional[str]`
```

Author: I ended up just removing this. Thanks for the suggestion!
